File: android/src/main/java/com/scandit/datacapture/flutter/core/utils/FlutterResult.java
/*
 * This file is part of the Scandit Data Capture SDK
 *
 * Copyright (C) 2024- Scandit AG. All rights reserved.
 */
package com.scandit.datacapture.flutter.core.utils;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.scandit.datacapture.frameworks.core.result.FrameworksResult;
import com.scandit.datacapture.frameworks.core.utils.DefaultFrameworksLog;
import com.scandit.datacapture.frameworks.core.utils.FrameworksLog;

import io.flutter.plugin.common.MethodChannel;

public class FlutterResult implements FrameworksResult {
    private final MethodChannel.Result result;

    public FlutterResult(MethodChannel.Result result) {
        this.result = result;
    }

    @Override
    public void success(@Nullable Object result) {
        this.result.success(result);
    }

    @Override
    public void error(@NonNull String errorCode, @Nullable String errorMessage, @Nullable Object errorDetails) {
        this.result.error(errorCode, errorMessage, errorDetails);
    }
}

File: android/src/main/kotlin/com/scandit/datacapture/flutter/core/utils/FlutterResult.kt
package com.scandit.datacapture.flutter.core.utils

import com.scandit.datacapture.frameworks.core.result.FrameworksResult
import com.scandit.datacapture.frameworks.core.utils.DefaultFrameworksLog
import com.scandit.datacapture.frameworks.core.utils.FrameworksLog
import io.flutter.plugin.common.MethodChannel

class FlutterResult(private val result: MethodChannel.Result) : FrameworksResult {
    override fun success(result: Any?) {
        this.result.success(result)
    }

    override fun error(errorCode: String, errorMessage: String?, errorDetails: Any?) {
        this.result.error(errorCode, errorMessage, errorDetails)
    }
}

class FlutterLogInsteadOfResult(
    private val logger: FrameworksLog = DefaultFrameworksLog.getInstance()
) : FrameworksResult {
    override fun success(result: Any?) {
        // nop
    }

    override fun error(errorCode: String, errorMessage: String?, errorDetails: Any?) {
        logger.error(
            "ErrorCode: $errorCode; ErrorMessage:$errorMessage; ErrorDetails:$errorDetails"
        )
    }
}
